src/MongoDB.Driver.Core/Core/Clusters/ClusterDescription.cs

```diff
@@ -65,10 +66,24 @@ internal static ClusterDescription CreateInitial(ClusterId clusterId, ClusterCon
             _connectionMode = connectionMode;
             _type = type;
             _servers = (servers ?? new ServerDescription[0]).OrderBy(n => n.EndPoint, new ToStringComparer<EndPoint>()).ToList();
-            _logicalSessionTimeout = CalculateLogicalSessionTimeout(_servers);
+
+            var serversThatDetermineWhetherSessionsAreSupported = SelectServersThatDetermineWhetherSessionsAreSupported(_servers).ToList();
+            _areSessionsSupported = CalculateAreSessionsSupported(serversThatDetermineWhetherSessionsAreSupported);
```

Comment:
Because ClusterDescriptions are immutable we can calculate whether sessions are supported once.

This is a minor performance enhancement (ClusterDescriptions don't change that often).